lib/Target/X86/X86FrameLowering.cpp

```diff
@@ -450,29 +450,25 @@ MachineInstr *X86FrameLowering::emitStackProbe(MachineFunction &MF,
 
 void X86FrameLowering::inlineStackProbe(MachineFunction &MF,
                                         MachineBasicBlock &PrologMBB) const {
-  MachineBasicBlock::iterator MBBI = PrologMBB.begin();
-  MachineBasicBlock::iterator MBBEnd = PrologMBB.end();
   const StringRef ChkStkStubSymbol = "__chkstk_stub";
   MachineInstr *ChkStkStub = nullptr;
 
-  for (; (MBBI != MBBEnd) && (ChkStkStub == nullptr); MBBI++) {
-    if (MBBI->isCall()) {
-      for (const auto &MO : MBBI->operands()) {
-        if (MO.isSymbol() && (ChkStkStubSymbol == MO.getSymbolName()))
-          ChkStkStub = MBBI;
-        break;
-      }
+  for (MachineInstr &MI : PrologMBB) {
+    if (MI.isCall() && MI.getOperand(0).isSymbol() &&
+        ChkStkStubSymbol == MI.getOperand(0).getSymbolName()) {
+      ChkStkStub = &MI;
+      break;
     }
   }
 
   if (ChkStkStub != nullptr) {
+    MachineBasicBlock::iterator MBBI = ChkStkStub->getIterator();
+    ++MBBI;
```

Comment:
Style nit: I like `MachineBasicBlock::iterator MBBI = std::next(ChkStkStub->getIterator());` for situations like this (simply for readability, to embed the throwaway intermediate state in the init expression).